examples/src/keyrings/raw_rsa_keyring_example.py

```diff
@@ -0,0 +1,191 @@
+with RSA public and private keys that you provide. In this example, we define the RSA keys to
+encrypt and decrypt the data keys.
+
+You need to generate, store, and protect the private key, preferably in a
+hardware security module (HSM) or key management system.
+The encryption function encrypts the data key under the RSA public key. The decryption function
+decrypts the data key using the private key.
+
+This example creates a Raw RSA Keyring and then encrypts a custom input EXAMPLE_DATA
+with an encryption context. This example also includes some sanity checks for demonstration:
+1. Ciphertext and plaintext data are not the same
+2. Encryption context is correct in the decrypted message header
+3. Decrypted plaintext value matches EXAMPLE_DATA
+4. After verifying that the encrypt and decrypt works, this example also demonstrates
+that the original ciphertext should not be decrypted using a new Raw RSA keyring generated by
+another user, let's say Bob (Points 9 and 10).
+These sanity checks are for demonstration in the example only. You do not need these in your code.
+
+A Raw RSA keyring that encrypts and decrypts must include an asymmetric public key and private
+key pair. However, you can encrypt data with a Raw RSA keyring that has only a public key,
+and you can decrypt data with a Raw RSA keyring that has only a private key. You can include
+any Raw RSA keyring in a multi-keyring. If you configure a Raw RSA keyring with a public and
+private key, be sure that they are part of the same key pair. Some language implementations
+of the AWS Encryption SDK will not construct a Raw RSA keyring with keys from different pairs.
+Others rely on you to verify that your keys are from the same key pair.
+
+For more information on how to use Raw RSA keyrings, see
+https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/use-raw-rsa-keyring.html
+"""
+import sys
+
+from aws_cryptographic_materialproviders.mpl import AwsCryptographicMaterialProviders
+from aws_cryptographic_materialproviders.mpl.config import MaterialProvidersConfig
+from aws_cryptographic_materialproviders.mpl.models import CreateRawRsaKeyringInput, PaddingScheme
+from aws_cryptographic_materialproviders.mpl.references import IKeyring
+from aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_materialproviders.errors import (
+    CollectionOfErrors,
+)
+from cryptography.hazmat.backends import default_backend as crypto_default_backend
+from cryptography.hazmat.primitives import serialization as crypto_serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from typing import Dict
+
+import aws_encryption_sdk
+from aws_encryption_sdk import CommitmentPolicy
+
+# TODO-MPL: Remove this as part of removing PYTHONPATH hacks.
+MODULE_ROOT_DIR = '/'.join(__file__.split("/")[:-1])
+
+sys.path.append(MODULE_ROOT_DIR)
+
+EXAMPLE_DATA: bytes = b"Hello World"
+
+
+def generate_rsa_keyring():
```

Comment:
Can we refactor this so the logic is

```
if user has provided private/public key PEMs:
    load those PEMs
otherwise:
    generate key pair, write PEM to file
load keyring from PEMs

```

We should allow users to bring their own PEMs to this example.
Most users who are interested in a raw RSA keyring will already have PEM-encoded files that they want to provide.
We should let them feed those keys into this example.
But, this also gives them some sample code that they can copy-paste to load their keys in their own code, rather than having to write load-from-file code themselves.

I see you're doing what we did in [Java-ESDK](https://github.com/aws/aws-encryption-sdk-java/blob/master/src/examples/java/com/amazonaws/crypto/examples/keyrings/RawRsaKeyringExample.java), but that example is not great.
The [NET-ESDK](https://github.com/aws/aws-encryption-sdk-dafny/blob/mainline/AwsEncryptionSDK/runtimes/net/Examples/Keyring/RawRSAKeyringExample.cs), we also doesn't do a good job of this; it writes stuff to files, but doesn't let users bring their own files out of the box.
The model here should be [Java-DBESDK](https://github.com/aws/aws-database-encryption-sdk-dynamodb/blob/main/Examples/runtimes/java/DynamoDbEncryption/src/main/java/software/amazon/cryptography/examples/keyring/RawRsaKeyringExample.java).

Feel free to push back if this is too much and we can find a middle ground